mitiq/mitiq_qiskit/conversions.py

```diff
@@ -46,6 +46,22 @@ def _remove_barriers(circuit: qiskit.QuantumCircuit) -> qiskit.QuantumCircuit:
     return copy
 
 
+def _remove_qasm_barriers(qasm: QASMType) -> QASMType:
+    """Returns a copy of the input QASM with all barriers removed.
+
+    Args:
+        qasm: QASM to remove barriers from.
+
+    Note:
+        According to the OpenQASM 2.X language specification
+        (https://arxiv.org/pdf/1707.03429v2.pdf), "Statements are separated by
+        semicolons. Whitespace is ignored. The language is case sensitive.
+        Comments begin with a pair of forward slashes and end with a new line."
+    """
+    # The inside regex removes comments, the outside regex removes barriers
```

Comment:
Why remove comments?

Reply:
Removing the comments is to remove all the weird things a comment can contain, for example, a semicolon, which will interfere with our matching of the barrier statements. Qiskit's qasm parser won't care about comments anyway.